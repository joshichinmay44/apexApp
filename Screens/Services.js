import React, {Component} from 'react';
import {
  StyleSheet,
  View,
  ScrollView,
  Image,
  TouchableOpacity,
} from 'react-native';
import {Appbar, Button, Text, TextInput, Card} from 'react-native-paper';
import styles from '../style/Style';

export default class Services extends Component {
  state={
    service:''
  }
  static navigationOptions = {
    headerShown: false,
  };

  infoArmy = () => {
    this.props.navigation.navigate('AboutTheForces',{
      service:'Indian Army'
    });
  };
  infoNavy = () => {
    this.props.navigation.navigate('AboutTheForces',{
      service:'Indian Navy'
    });
  };

  infoAirForce = () => {
    this.props.navigation.navigate('AboutTheForces',{
      service:'Indian Air Force'
    });
  };

  render() {
    return (
      <View style={styles.Container}>
       
          <Appbar.Header>
            <Appbar.Action
              icon="menu"
              onPress={() => this.props.navigation.openDrawer()}
            />
            <Appbar.Content title="Services" />
          </Appbar.Header>
          <ScrollView style={styles.Scroll}>
            <TouchableOpacity onPress={this.infoArmy}>
              <View style={{marginTop:50}}></View>
         <Card style={styles.cardContainer}>

          <Card.Cover source={require('../images/IndianArmyLogo.jpg')} />

         </Card>
         
        
         </TouchableOpacity>

         <TouchableOpacity onPress={this.infoNavy}>
              <View style={{marginTop:50}}></View>
         <Card style={styles.cardContainer}>

          <Card.Cover source={require('../images/IndianNavyLogo.png')} />

         </Card>
         
        
         </TouchableOpacity>

         <TouchableOpacity onPress={this.infoAirForce}>
              <View style={{marginTop:50}}></View>
         <Card style={styles.cardContainer}>

          <Card.Cover source={require('../images/IndianAirForce.jpg')} />

         </Card>
         
        
         </TouchableOpacity>
        </ScrollView>
      </View>
    );
  }
}
