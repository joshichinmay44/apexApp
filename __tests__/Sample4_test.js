import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Services from '../Screens/Services';
import {TestScheduler} from 'jest';

/* global.fetch = jest.fn(() => new Promise(resolve => resolve()));
jest.mock('react-native-gesture-handler', () => {});
jest.mock('react-native-reanimated', () => {});
jest.mock('@react-navigation/drawer', () => {});
jest.mock('@react-navigation/stack', () => {});
 */
test('renders correctly', () => {
  const snap = renderer.create(<Services />).toJSON();
  expect(snap).toMatchSnapshot();
});
