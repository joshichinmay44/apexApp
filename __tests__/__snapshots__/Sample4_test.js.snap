// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "ghostwhite",
      "height": "100%",
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "elevation": 4,
          "zIndex": 0,
        },
        Object {
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 3,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6200ee",
          "elevation": 0,
          "flexDirection": "row",
          "height": 56,
          "marginTop": 0,
          "paddingHorizontal": 4,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "margin": 6,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 18,
                "height": 36,
                "width": 36,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        <View
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                Ôçú
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            Array [
              Object {
                "marginLeft": 8,
              },
              Object {
                "alignItems": "center",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          accessible={true}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "fontSize": 17,
                },
                undefined,
              ],
            ]
          }
        >
          Services
        </Text>
      </View>
      <View
        style={
          Object {
            "width": 48,
          }
        }
      />
    </View>
  </RCTSafeAreaView>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "ghostwhite",
        "height": "100%",
        "marginBottom": 10,
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 50,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 4,
              "borderWidth": 1,
              "elevation": 1,
              "marginBottom": "5%",
              "marginTop": 3,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 0.75,
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#eeeeee",
                    "height": 195,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                  undefined,
                ]
              }
            >
              <Image
                siblings={
                  Array [
                    "withTheme(Card.Cover)",
                  ]
                }
                source={
                  Object {
                    "testUri": "../../../images/IndianArmyLogo.jpg",
                  }
                }
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "height": undefined,
                      "justifyContent": "flex-end",
                      "padding": 16,
                      "resizeMode": "cover",
                      "width": undefined,
                    },
                    Object {
                      "borderRadius": 4,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 50,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 4,
              "borderWidth": 1,
              "elevation": 1,
              "marginBottom": "5%",
              "marginTop": 3,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 0.75,
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#eeeeee",
                    "height": 195,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                  undefined,
                ]
              }
            >
              <Image
                siblings={
                  Array [
                    "withTheme(Card.Cover)",
                  ]
                }
                source={
                  Object {
                    "testUri": "../../../images/IndianNavyLogo.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "height": undefined,
                      "justifyContent": "flex-end",
                      "padding": 16,
                      "resizeMode": "cover",
                      "width": undefined,
                    },
                    Object {
                      "borderRadius": 4,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 50,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 4,
              "borderWidth": 1,
              "elevation": 1,
              "marginBottom": "5%",
              "marginTop": 3,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 0.5,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 0.75,
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#eeeeee",
                    "height": 195,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 4,
                  },
                  undefined,
                ]
              }
            >
              <Image
                siblings={
                  Array [
                    "withTheme(Card.Cover)",
                  ]
                }
                source={
                  Object {
                    "testUri": "../../../images/IndianAirForce.jpg",
                  }
                }
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "height": undefined,
                      "justifyContent": "flex-end",
                      "padding": 16,
                      "resizeMode": "cover",
                      "width": undefined,
                    },
                    Object {
                      "borderRadius": 4,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
